api/src/main/resources/AddressToEntryMap.hbm.xml

```diff
@@ -9,7 +9,9 @@
 
 		<id name="addressToEntryMapId" type="int" column="address_to_entry_map_id"
 			unsaved-value="0">
-			<generator class="native" />
+			<generator class="native">
+				<param name="sequence">address_hierarchy_address_to_entry__address_to_entry_map_id_seq</param>

```

Comment:
Is this name correct?

Reply:
Yes this will be the name since the table name strips by characters( max length is 63 characters) and "map" gets removed.

Reply:
Whenever you do that, can you always include a comment stating what the correct name should be and why you are changing it to this?

Reply:
Sure from now on I am adding all the changes in ticket description as well as in comments here.

Reply:
There is a problem in actual length of sequence that can be stored in PostgreSQL. When length excedes the limit then table name gets stripped from end.

Here there is stripping in one sequence only :

Original Sequence Name- address_hierarchy_address_to_entry_map_address_to_entry_map_id_seq

Current name as set in AddressToEntryMap.hbm file - address_hierarchy_address_to_entry__address_to_entry_map_id_seq

Reply:
In the code itself. Better to have comments as close to the code as possible.

Reply:
@dkayiwa , I have added the comments in code.

Reply:
Did you intentionally use `__` instead of `_`?

Reply:
The table name is "address_hierarchy_address_to_entry_map" , Now map gets sliced and then it becomes "address_hierarchy_address_to_entry_". Now according to the rules it will be concatenated with "_" and table id which is "address_to_entry_map_id" and then with "seq" . The format is tablename_tableidcolumn_seq. Hence there are two underscores.

Reply:
What happens when we just use one?

Reply:
Apparently the sequence that is being created in db is **address_hierarchy_address_to_entry__address_to_entry_map_id_seq** . Here is the link to screenshot on PostgreSQL db https://pasteboard.co/JjkBaM1.png . I executed the create table query on PostgreSQL for this.

Reply:
So we create a sequence name with one underscore and it creates another with two?

Reply:
We are never specifying the name of the sequence to be created anywhere. It is the sequence that creates by default when we set the id column type to serial in PostgreSQL. Liquibase does that internally for us. We are then mapping the created sequences with hibernate entities.

Reply:
Got you. 👍